### PR TITLE
Added support for Chinese handicap stone placement

### PR DESCRIPTION
**Pull request by roy7**

### CODE REVIEW

**Comment by roy7** [comment]
I pulled in your master version bump which your devel branch did not, need me to reverse that?

File: gtp2ogs.js [hunk 1]
```diff
@@ -554,23 +559,36 @@ class Game {
...
-                // Since the bot isn't restarting each move, we need to tell it about opponent moves
-                // Track and send each opponent move by tracking player colors.
-                //
+            // If handicap == 1, no extra stones are played.
+            // If we are black, we played after initial gamedata and so handicap is not < length.
+            // If we are white, this.state.moves.length will be 1 and handicap is not < length.
+            //
+            // If handicap >= 1, we don't check for opponent_evenodd to move on our turns until handicaps are finished.
+            //
+            this.log(this.state.free_handicap_placement, this.state.handicap, this.state.moves.length);
```
Comment (anoek) [inline gtp2ogs.js]: Is this log statement still useful, or should it be dropped?

Comment (roy7) [inline gtp2ogs.js]: Oops!

**Comment by roy7** [comment]
I need to test more. All testing on beta server worked, but game #7947185 hung during handicap placement.

**Comment by roy7** [comment]
Old Japanese handicap game with 2 stones had player order reversed when it unpaused. Will work on that.

**Comment by roy7** [comment]
Fixed evenodd for Japanese.

**Comment by anoek** [comment]
Awesome :)

**Pull request merged**